src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -16564,5 +16565,834 @@ public static void Main()
             var x2 = nodes.OfType<VariableDeclaratorSyntax>().Skip(1).First();
             Assert.Equal("<anonymous type: (System.Int32 c, System.Int32) Tuple> x2", model.GetDeclaredSymbol(x2).ToTestDisplayString());
         }
+
+        [Fact]
+        public void DefiniteAssignment001()
+        {
+            var source = @"
+using System;
+class C
+{
+        static void Main(string[] args)
+        {
+            (string A, string B) ss;
+
+            ss.A = ""q"";
+            ss.Item2 = ""w"";
+
+            System.Console.WriteLine(ss);
+        }
+    }
+";
+
+            var comp = CompileAndVerify(source,
+                additionalRefs: s_valueTupleRefs,
+                parseOptions: TestOptions.Regular, expectedOutput: @"
+(q, w)
+");
+        }
+
+        [Fact]
+        public void DefiniteAssignment002()
+        {
+            var source = @"
+using System;
+class C
+{
+        static void Main(string[] args)
+        {
+            (string A, string B) ss;
+
+            ss.A = ""q"";
+            ss.B = ""q"";
+            ss.Item1 = ""w"";
+            ss.Item2 = ""w"";
+
+            System.Console.WriteLine(ss);
+        }
+    }
+";
+
+            var comp = CompileAndVerify(source,
+                additionalRefs: s_valueTupleRefs,
+                parseOptions: TestOptions.Regular, expectedOutput: @"(w, w)");
+        }
+
+        [Fact]
+        public void DefiniteAssignment003()
+        {
+            var source = @"
+using System;
+class C
+{
+        static void Main(string[] args)
+        {
+            (string A, (string B, string C) D) ss;
+
+            ss.A = ""q"";
+            ss.D.B = ""w"";
+            ss.D.C = ""e"";
+
+            System.Console.WriteLine(ss);
+        }
+    }
+";
+
+            var comp = CompileAndVerify(source,
+                additionalRefs: s_valueTupleRefs,
+                parseOptions: TestOptions.Regular, expectedOutput: @"
+(q, (w, e))
+");
+        }
+
+        [Fact]
+        public void DefiniteAssignment004()
+        {
+            var source = @"
+using System;
+class C
+{
+        static void Main(string[] args)
+        {
+            (string I1, 
```

Comment:
Could you include a test where the friendly element names are "Item1", "Item2", ...?